pkg/proxy/backend/handshake_handler.go

```diff
@@ -28,13 +28,49 @@ const (
 	ConnContextKeyTLSState ConnContextKey = "tls-state"
 )
 
+type ErrorSource int
+
+const (
+	// SrcClientConn includes: client quit; bad client conn
+	SrcClientConn ErrorSource = iota
```

Comment:
I think `SrcClientConn` did not reveal its meaning. Maybe `XXXDisconnect` or `XXXQuit`?